core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -63,10 +63,15 @@ private[spark] class ChunkedByteBuffer(var chunks: Array[ByteBuffer]) {
    */
   def writeFully(channel: WritableByteChannel): Unit = {
     for (bytes <- getChunks()) {
-      while (bytes.remaining() > 0) {
-        val ioSize = Math.min(bytes.remaining(), bufferWriteChunkSize)
-        bytes.limit(bytes.position() + ioSize)
-        channel.write(bytes)
+      val curChunkLimit = bytes.limit()
+      while (bytes.hasRemaining) {
+        try {
+          val ioSize = Math.min(bytes.remaining(), bufferWriteChunkSize)
+          bytes.limit(bytes.position() + ioSize)
+          channel.write(bytes)
+        } finally {
```

Comment:
I don't think we need the `try` and `finally` here because `getChunks()` returns duplicated ByteBuffers which have their own position and limit.

Reply:
Do you mean this is not a real bug that can cause real workload to fail?

Reply:
I think the problem is, `bytes.limit(bytes.position() + ioSize)` will change the result of `bytes.hasRemaining`, so we have to restore the limit in each loop.

Reply:
I get your point. if there is an exception, there is no next loop and we don't need to restore the limit. so try finally is not needed